src/dataset/mod.rs

```diff
@@ -115,6 +115,14 @@ impl PartialOrd for Pr {
     }
 }
 
+impl Sub for Pr {
+    type Output = Pr;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Pr(self.0 - rhs.0)
```

Comment:
Should we check for negatives here?

Reply:
I agree but I'm not sure if it should be here. We can hide `f32` within `Pr` and instead use some method, e.g. `try_from` to create `Pr` while checking for negative. Something like that:
```rust
pub struct Pr(f32)

// and later initialize with
Pr::try_from(0.7)?
```

What do you think?

Reply:
The main drawback of doing this is that it breaks existing code that constructs `Pr`. While there's not much code doing that in `linfa`, there might be external users who'll be broken. However, `Pr` is basically useless if we don't restrain its value, and we already have breaking changes in this release. I'd say add a `try_from` impl that checks `0 <= prob <= 1.0` and returns `Result<Pr, f32>`. Then add `new`, which constructs `Pr` infallibly and panics on bad input (this is how `Pr` will be constructed most of the time). Finally add `new_unchecked` which constructs `Pr` without any checking, for converting large batches. You'll need to add the invariant of `0 <= prob <= 1.0` to `Pr`'s docs, and get rid of the `Sub` impl.

If you really want to keep the `Sub` impl, then just panic in the `Sub` impl if the result is under 0. Regardless of how you approach this, I'd like you to add `#[repr(transparent)]` to `Pr`.